flink-connectors/flink-connector-elasticsearch-base/src/test/java/org/apache/flink/connector/elasticsearch/table/IndexGeneratorTest.java

```diff
@@ -299,67 +300,58 @@ public void testGeneralDynamicIndex() {
     }
 
     @Test
     public void testStaticIndex() {
         IndexGenerator indexGenerator =
                 IndexGeneratorFactory.createIndexGenerator("my-index", fieldNames, dataTypes);
         indexGenerator.open();
-        Assertions.assertEquals("my-index", indexGenerator.generate(rows.get(0)));
-        Assertions.assertEquals("my-index", indexGenerator.generate(rows.get(1)));
+        assertThat(indexGenerator.generate(rows.get(0))).isEqualTo("my-index");
+        assertThat(indexGenerator.generate(rows.get(1))).isEqualTo("my-index");
     }
 
     @Test
     public void testUnknownField() {
         String expectedExceptionMsg =
                 "Unknown field 'unknown_ts' in index pattern 'my-index-{unknown_ts|yyyy-MM-dd}',"
                         + " please check the field name.";
-        try {
-            IndexGeneratorFactory.createIndexGenerator(
-                    "my-index-{unknown_ts|yyyy-MM-dd}", fieldNames, dataTypes);
-        } catch (TableException e) {
-            Assertions.assertEquals(e.getMessage(), expectedExceptionMsg);
-        }
+        assertThatThrownBy(
+                        () ->
+                                IndexGeneratorFactory.createIndexGenerator(
+                                        "my-index-{unknown_ts|yyyy-MM-dd}", fieldNames, dataTypes))
+                .isInstanceOf(TableException.class)
+                .hasMessage(expectedExceptionMsg);
     }
 
     @Test
     public void testUnsupportedTimeType() {
         String expectedExceptionMsg =
                 "Unsupported type 'INT' found in Elasticsearch dynamic index field, "
                         + "time-related pattern only support types are: DATE,TIME,TIMESTAMP.";
-        try {
-            IndexGeneratorFactory.createIndexGenerator(
-                    "my-index-{id|yyyy-MM-dd}", fieldNames, dataTypes);
-        } catch (TableException e) {
-            Assertions.assertEquals(expectedExceptionMsg, e.getMessage());
-        }
+        assertThatThrownBy(
+                        () ->
+                                IndexGeneratorFactory.createIndexGenerator(
+                                        "my-index-{id|yyyy-MM-dd}", fieldNames, dataTypes))
+                .isInstanceOf(TableException.class)
+                .hasMessage(expectedExceptionMsg);
     }
 
     @Test
     public void testUnsupportedMultiParametersType() {
         String expectedExceptionMsg =
                 "Chaining dynamic index pattern my-index-{local_date}-{local_time} is not supported,"
                         + " only support single dynamic index pattern.";
-        try {
-            IndexGeneratorFactory.createIndexGenerator(
-                    "my-index-{local_date}-{local_time}", fieldNames, dataTypes);
-        } catch (TableException e) {
-            Assertions.assertEquals(expectedExceptionMsg, e.getMessage());
-        }
-    }
-
-    @Test
-    public void testDynamicIndexUnsupportedFormat() {
```

Comment:
This test got removed. Is that intentional?

Reply:
Yes, I removed the test because the Exception is not being thrown in the current implementation

Reply:
I am not sure I get it. I thought this PR is only supposed to change the tests, not the implementation. How was the test passing if the exception is not thrown?

Reply:
The old test used try-catch and asserted on the assertion if it was caught. However, as there is no assertion being thrown it would always pass. This is different now because assertThatThrownBy actually causes the test to fail if no exception is thrown